File: parsers/go/parser.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

type Request struct {
	proto   string
	version string
	command string
	headers []byte
	body    []byte
}

func reqParse(req string) Request {
	res := strings.Split(req, "\n")

	reqLine := res[0]
	payload := strings.Join(res[1:], "\n")
	r := strings.Split(reqLine, " ")
	if len(r) < 8 {
		r = append(r, " ")
	}
	headersLen, _ := strconv.Atoi(r[3])
	bodyLen, _ := strconv.Atoi(r[4])
	headers := payload[0:headersLen]
	body := payload[headersLen : headersLen+bodyLen]

	request := Request{
		proto:   r[0],
		version: r[1],
		command: r[2],
		headers: []byte(headers),
		body:    []byte(body),
	}
	return request
}

func reqMarshal(req Request) string {
	requestLine := fmt.Sprintf("%s %s %s %d %d",
		req.proto,
		req.version,
		req.command,
		len(req.headers),
		len(req.body))

	r := fmt.Sprintf("%s\n%s%s", requestLine, string(req.headers), string(req.body))
	return r
}

File: parsers/python/parser.py
#!/usr/bin/env python

class Request:
    def __init__(self, proto, version, command, headers, body):
        self.proto = proto
        self.version = version
        self.command = command
        self.headers = headers
        self.body = body

def req_parse(req):
    res = req.split("\n",1)
    reqLine = res[0]
    tmp = res
    for i in range (0,len(res)):
        if tmp[i] == "":
            tmp[i] = "\n"
    payload = "".join(tmp[1:])
    r = str.split(reqLine, " ")
    if len(r) < 8:
        r.append("")
    headersLen = int(r[3])
    bodyLen = int(r[4])
    headers = payload[0:headersLen]
    body = payload[headersLen:bodyLen+headersLen]

    request = Request(r[0],r[1],r[2],headers,body)
    return request.__dict__

def req_marshal(req):
    if isinstance(req,dict):
        proto = req.get("proto")
        version = req.get("version")
        command = req.get("command")
        headerLen = len(req.get("headers"))
        bodyLen = len(req.get("body"))
        headers = (req.get("headers"), "")[headerLen == 0]
        body = (req.get("body"), "")[bodyLen == 0]
        return ("{} {} {} {} {}\n{}{}").format(proto, version, command, headerLen, bodyLen, headers, body)
